### PR TITLE
[dockers] update mellanox syncd and pmon to buster

### PR DESCRIPTION
**Pull request by stepanblyschak**
Signed-off-by: Stepan Blyschak <stepanb@mellanox.com>

**- Why I did it**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by lguohan** [comment]
looks like some build failure

**Comment by stepanblyschak** [comment]
@lguohan mellanox build requires https://github.com/Mellanox/Switch-SDK-drivers/pull/17, but i shared PR right now so you are able to test buster pmon on non mellanox platforms

**Comment by liat-grozovik** [comment]
Retest mellanox please

**Comment by stepanblyschak** [comment]
@lguohan 
There is an issue with ctypesgen==0.r125 during saithrift compilation:
```
Status: gcc -E -U __GNUC__ -dD -I/usr/include -I/usr/include/sai "-Dinline=" "-D__inline__=" "-D__extension__=" "-D_Bool=uint8_t" "-D__const=const" "-D__asm__(x)=" "-D__asm(x)=" "-DCTYPESGEN=1" /tmp/tmpAR8Dwz.h
Traceback (most recent call last):
  File "/usr/local/bin/ctypesgen.py", line 136, in <module>
    descriptions=ctypesgencore.parser.parse(options.headers,options)
  File "/usr/local/lib/python2.7/dist-packages/ctypesgencore/parser/__init__.py", line 21, in parse
    parser.parse()
  File "/usr/local/lib/python2.7/dist-packages/ctypesgencore/parser/datacollectingparser.py", line 69, in parse
    ctypesparser.CtypesParser.parse(self, fname, None)
  File "/usr/local/lib/python2.7/dist-packages/ctypesgencore/parser/cparser.py", line 113, in parse
    self.preprocessor_parser.parse(filename)
  File "/usr/local/lib/python2.7/dist-packages/ctypesgencore/parser/preprocessor.py", line 200, in parse
    token = self.lexer.token()
  File "/usr/local/lib/python2.7/dist-packages/ctypesgencore/parser/preprocessor.py", line 51, in token
    result = lex.Lexer.token(self)
  File "/usr/local/lib/python2.7/dist-packages/ctypesgencore/parser/lex.py", line 297, in token
    newtok = func(tok)
  File "/usr/local/lib/python2.7/dist-packages/ctypesgencore/parser/pplexer.py", line 218, in t_ANY_int
    g1 = str(long(g1,8))
ValueError: invalid literal for long() with base 8: '08420217248550443400745280086994171'
```

Updated ctypesgen to 1.0.2, but it requires this fix in SAI https://github.com/opencomputeproject/SAI/pull/1111

**Comment by stepanblyschak** [comment]
retest mellanox please

File: platform/template/docker-syncd-buster-base.mk [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# docker image for syncd
+
+
+DOCKER_SYNCD_BASE_STEM = docker-syncd-$(DOCKER_SYNCD_PLATFORM_CODE)
```
Comment (lguohan) [inline platform/template/docker-syncd-buster-base.mk:4]: docker-syncd-base.mk has been adapted to support buster, this is not needed.

File: platform/mellanox/docker-syncd-mlnx.mk [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # docker image for mlnx syncd
 
 DOCKER_SYNCD_PLATFORM_CODE = mlnx
-include $(PLATFORM_PATH)/../template/docker-syncd-base.mk
+include $(PLATFORM_PATH)/../template/docker-syncd-buster-base.mk
```
Comment (lguohan) [inline platform/mellanox/docker-syncd-mlnx.mk:4]: checkout docker-syncd-vs.mk it is already buster now.

**Comment by stepanblyschak** [comment]
retest broadcom please

**Comment by stepanblyschak** [comment]
retest vsimage please

**Comment by akokhan** [comment]
@stepanblyschak this PR has updated sairedis without even mentioning that... at least sairedis commits should be listed in the description. Thanks.

**Pull request merged**